src/vs/editor/contrib/stickyScroll/browser/stickyScroll.ts

```diff
@@ -66,12 +67,24 @@ class StickyScrollController extends Disposable implements IEditorContribution {
 			this._editor.addOverlayWidget(this.stickyScrollWidget);
 			this._sessionStore.add(this._editor.onDidChangeModel(() => this._update(true)));
 			this._sessionStore.add(this._editor.onDidScrollChange(() => this._update(false)));
+			this._sessionStore.add(this._editor.onDidChangeModelTokens((e) => this._onTokensChange(e)));
 			this._sessionStore.add(this._editor.onDidChangeModelContent(() => this._updateSoon.schedule()));
 			this._sessionStore.add(this._languageFeaturesService.documentSymbolProvider.onDidChange(() => this._update(true)));
 			this._update(true);
 		}
 	}
 
+	private _onTokensChange(event: IModelTokensChangedEvent) {
+		const beginningLine = this._editor.getVisibleRanges()[0].getStartPosition().lineNumber - 1;
+		const endLine = beginningLine + this.stickyScrollWidget.codeLineCount + 1;
+		const fromLineNumber = event.ranges[0].fromLineNumber;
+		const toLineNumber = event.ranges[0].toLineNumber;
+		if (fromLineNumber > endLine || beginningLine > toLineNumber) {
+			return;
+		}
```

Comment:
Here we should intersect all ranges present in `event.ranges`, not just `event.ranges[0]` with all the currently rendered `StickyScrollCodeLine`. 

The `StickyScrollWidget` could have a getter like `.getCurrentStickyLineNumbers(): number[]` which would return the line numbers currently rendered at the top and then that array would be intersected with each of the ranges in `event.ranges`

Reply:
Right, I'll change that